### PR TITLE
The endpoints /metadata/all and metadata/data-structures now accept the

### PR DESCRIPTION
**Pull request by essbb**
optional parameter skip_code_lists that will filter (i.e. set the list to empty) the following list items: skip_code_lists, missingValues. Note that the default value is false which preserves the existing behaviour

### CODE REVIEW

File: metadata_service/domain/metadata.py [hunk 1]
```diff
@@ -67,3 +71,34 @@ def find_languages():
...
+def find_all_metadata_skip_code_list_and_missing_values(version):
+    metadata = datastore.get_metadata_all(version)
+     
+    if 'dataStructures' in metadata:
+        __clear_code_list_and_missing_values(metadata['dataStructures'])
+    else:
+        __clear_code_list_and_missing_values(metadata)
+    return metadata
+
+def __clear_code_list_and_missing_values(metadata):
```
Comment (DanielElisenberg) [inline metadata_service/domain/metadata.py]: This looks good :100:

File: tests/unit/domain/test_metadata.py [hunk 2]
```diff
@@ -164,3 +170,51 @@ def test_find_all_datastore_versions_when_draft_version_empty(mocker):
...
+    
+    with open(DATA_STRUCTURES_FILE_PATH, encoding='utf-8') as f:
+        mocked_data_structures = json.load(f)
+
+    mocker.patch.object(
+        datastore, 'get_metadata_all',
+        return_value=mocked_data_structures
+    )
+    
+    filtered_metadata = metadata.find_all_metadata_skip_code_list_and_missing_values(version='1.0.0.0')
```
Comment (DanielElisenberg) [inline tests/unit/domain/test_metadata.py:192]: I would prefer an expected metadata_all_no_code_list__1_0_0_0.json file to compare to. Easier to see the expected output this way imho.

**Review by DanielElisenberg — CHANGES_REQUESTED** [review]
The logic looks good :100: Small request for change in testing

File: tests/unit/domain/test_metadata.py (hunk 2 above)
Comment (essbb) [reply tests/unit/domain/test_metadata.py:192]: Not sure what you're asking here, please clarify

Comment (essbb) [reply tests/unit/domain/test_metadata.py:192]: I suppose we should have a positive test that checks for the removal of these lists and perhaps another that checks that we don't botch a file that already a-ok ?

Comment (essbb) [reply tests/unit/domain/test_metadata.py:192]: Implemented

File: metadata_service/domain/metadata.py (hunk 1 above)
Comment (pawbu) [inline metadata_service/domain/metadata.py]: Use single underscore for indication of "internal use". [Ref](https://peps.python.org/pep-0008/#descriptive-naming-styles)

File: tests/unit/domain/test_metadata.py [hunk 3]
```diff
@@ -171,31 +178,41 @@ def test_find_all_datastore_versions_when_draft_version_empty(mocker):
...
+    with open(DATA_STRUCTURES_FILE_PATH, encoding='utf-8') as f:
+        mocked_data_structures = json.load(f)
 
     mocker.patch.object(
         datastore, 'get_metadata_all',
-        return_value=mocked_metadata_all
+        return_value=mocked_data_structures
     )
     
     filtered_metadata = metadata.find_all_metadata_skip_code_list_and_missing_values(version='1.0.0.0')
```
Comment (DanielElisenberg) [inline tests/unit/domain/test_metadata.py]: Am I reading this wrong or is this now only returning the metadata_all['dataStructures']-list? Is it not supposed to return the full metadata object? 🤔

File: tests/unit/domain/test_metadata.py [hunk 4]
```diff
@@ -164,3 +177,61 @@ def test_find_all_datastore_versions_when_draft_version_empty(mocker):
...
+    
+    filtered_metadata = metadata.find_all_metadata_skip_code_list_and_missing_values(version='1.0.0.0')
+    __assert_code_list_and_missing_values(filtered_metadata)
+
+    with open(DATA_STRUCTURES_NO_CODE_LIST_FILE_PATH, encoding='utf-8') as f:
+        data_structures_no_code_list = json.load(f)
+
+    assert data_structures_no_code_list == filtered_metadata
+
+def __assert_code_list_and_missing_values(metadata):
```
Comment (pawbu) [inline tests/unit/domain/test_metadata.py:213]: Use single underscore for indication of "internal use". [Ref](https://peps.python.org/pep-0008/#descriptive-naming-styles)

**Review by pawbu — APPROVED** [review]
Looks good, minor comment

File: tests/unit/domain/test_metadata.py (hunk 3 above)
Comment (essbb) [reply tests/unit/domain/test_metadata.py]: Could you please run the tests locally, it seems fine to me.
What may be confusing is that metadata_all wraps all the other stuff under 'dataStructures'  while this level does not exist in data structures. I've also tested with postman with the expected outcome. (will double check ASAP though)

File: tests/unit/domain/test_metadata.py (hunk 4 above)
Comment (essbb) [reply tests/unit/domain/test_metadata.py:213]: Ok

File: metadata_service/domain/metadata.py (hunk 1 above)
Comment (essbb) [reply metadata_service/domain/metadata.py]: Thanks, will fix

**Review by DanielElisenberg — APPROVED** [review]
Looks good :100:

**Pull request merged**